### DISCUSSION TITLE
Connecting to model.bim

### DISCUSSION THREAD

**Discussion by dotnjet** [DaxStudio/DaxStudio#899] [Q&A] [answered]
Thank you for this great tool. It is very beneficial and we have set the DAX Formatter as the standard in our company. 
Now we would like to parse all our model.bim files and like to automatically change the formatting of the DAX code to short DAX formatting. Is there a way to do it and how can we connect to the model.bim files?

**Comment by marcosqlbi** [comment, marked as answer]
You have several options:
-  [Bravo for Power BI](https://www.sqlbi.com/tools/bravo-for-power-bi/) 
- a script in [Tabular Editor](https://docs.tabulareditor.com/te2/FormatDax.html) 
- create your own code that uses the [DAX Formatter service](https://www.nuget.org/packages/Dax.Formatter/)

**Reply by dotnjet** [reply]
Hello Marco

Thanks for your very fast answer.
I have installed Bravo since your released it and I like it as well very much. How can I connect Bravo to a model.bim file?

Best Regards

Martin

**Reply by dgosbell** [reply]
DAX Studio always connects to models in a read-only manner. It does not make any changes to the data models so the options Marco suggested are the best way of achieving your goal

**Comment by marcosqlbi** [comment]
You cannot - so you should use one of the other two options!

**Reply by dotnjet** [reply]
Too bad. Let's add it as a feature request ;-)
